lxc/storage_volume.go

```diff
@@ -1113,38 +1223,127 @@ func (c *cmdStorageVolumeList) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	// Process the columns
+	columns, err := c.parseColumns(resource.server.IsClustered())
+	if err != nil {
+		return err
+	}
+
+	// Render the table
 	data := [][]string{}
-	for _, volume := range volumes {
-		usedby := strconv.Itoa(len(volume.UsedBy))
+	for _, vol := range volumes {
+		row := []string{}
+		for _, column := range columns {
+			if column.NeedsState && !shared.IsSnapshot(vol.Name) && vol.Type != "image" {
+				state, err := resource.server.GetStoragePoolVolumeState(resource.name, vol.Type, vol.Name)
+				if err != nil {
+					return err
+				}
+
+				row = append(row, column.Data(vol, *state))
+			} else {
+				row = append(row, column.Data(vol, api.StorageVolumeState{}))
+			}
+		}
+		data = append(data, row)
+	}
+	if len(columns) >= 2 {
+		sort.Sort(byNameAndType(data))
+	}
+
+	rawData := make([]*api.StorageVolume, len(volumes))
+	for i := range volumes {
+		rawData[i] = &volumes[i]
+	}
+
+	headers := []string{}
+	for _, column := range columns {
+		headers = append(headers, column.Name)
+	}
+
+	return utils.RenderTable(c.flagFormat, headers, data, rawData)
+}
+
+func (c *cmdStorageVolumeList) parseColumns(clustered bool) ([]volumeColumn, error) {
+	columnsShorthandMap := map[rune]volumeColumn{
+		't': {i18n.G("TYPE"), c.typeColumnData, false},
+		'n': {i18n.G("NAME"), c.nameColumnData, false},
+		'd': {i18n.G("DESCRIPTION"), c.descriptionColumnData, false},
+		'c': {i18n.G("CONTENT-TYPE"), c.contentTypeColumnData, false},
+		'u': {i18n.G("USED BY"), c.usedByColumnData, false},
+		'U': {i18n.G("USAGE"), c.usageColumnData, true},
+	}
 
-		entry := []string{volume.Type, volume.Name, volume.Description, volume.ContentType, usedby}
-		if shared.IsSnapshot(volume.Name) {
-			entry[0] = fmt.Sprintf("%s (snapshot)", volume.Type)
+	if clustered {
+		columnsShorthandMap['L'] = volumeColumn{
+			i18n.G("LOCATION"), c.locationColumnData, false}
```

Comment:
@stgraber please can you change this to initialise the new struct using specified field names e.g.

 `volumeColumn{Name: i18n.G("LOCATION"), Data: c.locationColumnData, NeedsState: false}`

Rather than relying on the field order. I understand that this is the recommended style nowadays, and in this case it slowed me down in finding where the `NeedsState` field was set as I could only find references to it being accessed not set.